DevTools/UserspaceEmulator/SoftCPU.h

```diff
@@ -1154,4 +1157,15 @@ ALWAYS_INLINE u32 SoftCPU::read32()
     return value;
 }
 
+ALWAYS_INLINE u64 SoftCPU::read64()
+{
+    if (!m_cached_code_ptr || (m_cached_code_ptr + 8) >= m_cached_code_end)
```

Comment:
The `+ 8` here is consistent with `read16()` and `read32()`, but not with `read8()`. I believe `read8()` is right and `read16()`, `read32()` and `read64()` should add 1, 3, and 7 instead of 2, 4, and 8. But that should probably be in a separate commit and even a separate pull request, and it conflicts with the one here, so I'd prefer to do it after this one is merged. But if you prefer I can send that out first, wait for that to be merged, and then rebase this on the other one.